src/utils/backend/vatsim/update.ts

```diff
@@ -245,11 +245,59 @@ export async function updateVatsimExtendedPilots() {
             extendedPilot.status = 'enroute';
         }
 
+        // Diversion Detection
+        if (extendedPilot.status === 'departed' || extendedPilot.status === 'climbing' ||
+            extendedPilot.status === 'cruising' || extendedPilot.status === 'enroute' ||
+            extendedPilot.status === 'descending' || extendedPilot.status === 'arriving') {
+
+            const oldPilot = radarStorage.vatsim.extendedPilots.find(x => x.cid === extendedPilot.cid);
```

Comment:
This is not very good for performance. Since we find pilot just by cid, I suggest to make a map/object somewhere above, where "key" will be the cid.

And this code will look like this

```suggestion
            const oldPilot = keyedExtendedPilots[extendedPilot.cid];
```